presto-main/src/main/java/com/facebook/presto/spiller/FileSingleStreamSpillerFactory.java

```diff
@@ -138,7 +139,7 @@ private static void cleanupOldSpillFiles(Path path)
     }
 
     @Override
-    public SingleStreamSpiller create(List<Type> types, SpillContext spillContext, LocalMemoryContext memoryContext)
+    public SingleStreamSpiller create(List<Type> types, Supplier<SpillContext> spillContext, LocalMemoryContext memoryContext)
```

Comment:
I do not like it.

It makes the signature more complex. Actually it used to be somewhat similar to what you propose originally.
Maybe good enough solution if we are afraid that someone calls `close()` on passed `spillContext`, is to create new child spill context at calling place.